rig-core/src/vector_store/mod.rs

```diff
@@ -17,6 +17,9 @@ pub enum VectorStoreError {
 
     #[error("Datastore error: {0}")]
     DatastoreError(#[from] Box<dyn std::error::Error + Send + Sync>),
+
+    #[error("Vector store error: {0}")]
+    Error(String),
```

Comment:
I'd use `DatastoreError` instead of creating a new error type here.

FYI `String` implements `Into<Box<dyn std::error::Error + Send + Sync>>`, so you can do `DatastoreError(format!("some error {some_stuff}").into())`